python/src/iceberg/literals.py

```diff
@@ -0,0 +1,595 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import datetime
+import uuid
+from decimal import ROUND_HALF_UP, Decimal
+
+import pytz
+
+from .types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    UUIDType,
+)
+
+
+class Literals(object):
```

Comment:
I'm going to spend more time looking at this but is there a reason why we need both `Literal` and `BaseLiteral`? Could what's here be collapsed into `BaseLiteral`?

Reply:
after some looking, I believe this is the hierarchy graph

```
Literal
|-- AboveMax
|-- BelowMin
|-- BaseLiteral
    |-- StringLiteral
    |-- FixedLiteral    
    |-- BinaryLiteral        
    |-- ComparableLiteral
        |-- BooleanLiteral
        |-- IntegerLiteral
        |-- LongLiteral
        |-- FloatLiteral
        |-- DoubleLiteral
        |-- DateLiteral
        |-- TimeLiteral
        |-- TimestampLiteral
        |-- DecimalLiteral
        |-- UUIDLiteral
```